kolibri/auth/test/test_api.py

```diff
@@ -417,3 +417,22 @@ def test_facility_user_cannot_delete_dataset(self):
         self.client.login(username=self.user.username, password=DUMMY_PASSWORD)
         response = self.client.delete(reverse('facilitydataset-detail', kwargs={'pk': self.facility.dataset_id}), format="json")
         self.assertEqual(response.status_code, 403)
+
+
+class MembershipCascadeDeletion(APITestCase):
+
+    def setUp(self):
+        provision_device()
+        self.facility = FacilityFactory.create()
+        self.superuser = create_superuser(self.facility)
+        self.user = FacilityUserFactory.create(facility=self.facility)
+        self.classroom = ClassroomFactory.create(parent=self.facility)
+        self.lg = LearnerGroupFactory.create(parent=self.classroom)
+
+    def test_delete_classroom_membership(self):
```

Comment:
Can you add a test where we DELETE at the membership-detail URL? I think it might make sense to change the front-end code to not do `MembershipResource.getCollection(...).delete()` after your change.